client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-assembler-preview.tsx

```diff
@@ -0,0 +1,98 @@
+import { getDesignPreviewUrl } from '@automattic/design-picker';
+import { useLocale } from '@automattic/i18n-utils';
+import { Spinner } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import classnames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useState } from 'react';
+import ReactDOM from 'react-dom';
+import WebPreview from 'calypso/components/web-preview/content';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { useSite } from '../../../../hooks/use-site';
+import { ONBOARD_STORE } from '../../../../stores';
+import PreviewToolbar from '../design-setup/preview-toolbar';
+import { encodePatternId } from './utils';
+import type { Pattern } from './types';
+import type { Design } from '@automattic/design-picker';
+import './pattern-assembler-preview.scss';
+
+interface Props {
+	header: Pattern | null;
+	sections?: Pattern[];
+	footer: Pattern | null;
+}
+
+const PatternAssemblerPreview = ( { header, sections = [], footer }: Props ) => {
+	const locale = useLocale();
+	const translate = useTranslate();
+	const site = useSite();
+	const [ webPreviewFrameContainer, setWebPreviewFrameContainer ] = useState< Element | null >(
+		null
+	);
+
+	const hasSelectedPatterns = header || sections.length > 0 || footer;
+	const selectedDesign = useSelect( ( select ) => select( ONBOARD_STORE ).getSelectedDesign() );
+
+	const mergedDesign = {
+		...selectedDesign,
+		recipe: {
+			...selectedDesign?.recipe,
+			// The blank canvas blocks demo site doesn't have the header, so we inject the header into the first pattern
+			// of the content.
+			pattern_ids: [ header, ...sections ]
+				.filter( Boolean )
+				.map( ( pattern ) => encodePatternId( pattern!.id ) ),
+			footer_pattern_ids: footer ? [ encodePatternId( footer.id ) ] : undefined,
+		},
+	} as Design;
+
+	useEffect( () => {
+		setWebPreviewFrameContainer( document.querySelector( '.web-preview__frame-wrapper' ) );
```

Comment:
For how much `<WebPreview/>` is used, it needs a revamp 🥲

Reply:
Yep. But it might be easy to show the empty placeholder when we're able to render the blocks on calypso

Reply:
I see what you mean, we can add a prop `blankUrlMessage` or `noPreviewMessage` to `<WebPreview/>`. I think using a portal, for now, is a good workaround for this specific case. 

I guess for feature requests we can just create an issue tagged with the component name.

We could also update the [docs](https://github.com/Automattic/wp-calypso/tree/trunk/client/components/web-preview). I'm thinking about adding docs for `<ThemePreview>` and moving it to the [components](https://github.com/Automattic/wp-calypso/tree/trunk/client/components) folder. What do you think?

Reply:
If we want, I'd prefer to move the `ThemePreview` into `@automattic/components`. It will be more general to use.